INSTALL_OPAM.md should be updated to include info about OCaml versions

I am currently installing UniMath on a new macOS system from scratch, using instructions from `INSTALL_OPAM.md`. In addition to things mentioned in the docs I had to install a couple of packages with Homebrew:

```sh
brew install pkg-config gtksourceview libxml2
```

Now I am trying to `make` UniMath, but am getting the following error:

```
...
make[2]: Entering directory `/Users/.../UniMath/sub/coq'
COQMKTOP -o bin/coqtop.opt
Undefined symbols for architecture x86_64:
  "_caml_young_limit", referenced from:
      _coq_interprete in libcoqrun.a(coq_interp.o)
  "_caml_young_ptr", referenced from:
      _coq_interprete in libcoqrun.a(coq_interp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
File "caml_startup", line 1:
Error: Error during linking
make[2]: *** [bin/coqtop.opt] Error 2
make[2]: Leaving directory `/Users/.../UniMath/sub/coq'
make[1]: *** [submake] Error 2
make[1]: Leaving directory `/Users/.../UniMath/sub/coq'
make: *** [sub/coq/bin/coq_makefile] Error 2
```

According to a message found [in the Gitter channel for Coq](https://gitter.im/coq/coq?at=5e7db28510235831bf02664a) the problem is that I have too fresh ocaml installed:

<img width="1083" alt="Screenshot 2020-04-13 at 20 22 15" src="https://user-images.githubusercontent.com/686582/79142635-78aeb900-7dc4-11ea-84b6-e507d58762a3.png">

And indeed, I have OCaml 4.10:

```
$ ocamlc --version
4.10.0
```

[A StackOverflow answer](https://stackoverflow.com/questions/61023002/coq-8-11-0-incompatible-with-ocaml-4-10) mentions (linking [a Discourse group post](https://coq.discourse.group/t/install-notes-on-coq-and-ocaml-versions-configuration/713)) that it is recommended to use OCaml 4.07.1+flambda. So I have switched to that:

```sh
opam switch create with-coq 4.07.1+flambda
```

And then proceeding from the corresponding place in `INSTALL_OPAM.md`:

```sh
opam install -y num lablgtk conf-gtksourceview lablgtk3-sourceview3 camlp5
```

After this I went to UniMath, cleaned up the repository and run

```
make
```

It is still going, but the problem seems to be gone (at least the process moved past the problematic point).

If everything goes well for my system, I suggest updating `INSTALL_OPAM.md` to recommend using a specific OCaml version or at least to include instructions to switch to a different version (I will make a PR).

I started from scratch, but with opamroot in a private location, so it has to recompile everything, and I don't even get as far as you do: the opam install step fails while trying to compile ocamlfind:
```
einsteinium$ opam install -y num lablgtk conf-gtksourceview lablgtk3-sourceview3 camlp5
[NOTE] Package conf-gtksourceview is already installed (current version is 2).
The following actions will be performed:
  ∗ install dune                 2.5.0   [required by lablgtk3-sourceview3]
  ∗ install ocamlfind            1.8.1   [required by lablgtk, num]
  ∗ install camlp5               7.11
  ∗ install dune-private-libs    2.5.0   [required by dune-configurator]
  ∗ install num                  1.3
  ∗ install lablgtk              2.18.10
  ∗ install dune-configurator    2.5.0   [required by cairo2]
  ∗ install cairo2               0.6.1   [required by lablgtk3]
  ∗ install lablgtk3             3.1.0   [required by lablgtk3-sourceview3]
  ∗ install lablgtk3-sourceview3 3.1.0
===== ∗ 10 =====

<><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫 
[cairo2.0.6.1] found in cache
[camlp5.7.11] found in cache
[dune.2.5.0] found in cache
[dune-configurator.2.5.0] found in cache
[dune-private-libs.2.5.0] found in cache
[lablgtk.2.18.10] found in cache
[lablgtk3.3.1.0] found in cache
[lablgtk3-sourceview3.3.1.0] found in cache
[num.1.3] found in cache
[ocamlfind.1.8.1] found in cache

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫 
[ERROR] The compilation of dune failed at "/Users/dan/.opamroot/opam-init/hooks/sandbox.sh
        build ocaml bootstrap.ml -j 7".
[ERROR] The compilation of camlp5 failed at "/Users/dan/.opamroot/opam-init/hooks/sandbox.sh
        build make world.opt".
[ERROR] The compilation of ocamlfind failed at
        "/Users/dan/.opamroot/opam-init/hooks/sandbox.sh build make all".

#=== ERROR while compiling ocamlfind.1.8.1 ====================================#
# context              2.0.6 | macos/x86_64 |  | https://opam.ocaml.org#e2f2c6d2
# path                 ~/.opamroot/empty/.opam-switch/build/ocamlfind.1.8.1
# command              ~/.opamroot/opam-init/hooks/sandbox.sh build make all
# exit-code            2
# env-file             ~/.opamroot/log/ocamlfind-99615-52d546.env
# output-file          ~/.opamroot/log/ocamlfind-99615-52d546.out
### output ###
# [...]
# fi
# ocamldep *.ml *.mli >depend
# ocamlc -I +compiler-libs -opaque -g -c findlib_config.ml
# ocamlc -I +compiler-libs -opaque -g -c fl_split.ml
# ocamlc -I +compiler-libs -opaque -g -c fl_metatoken.ml
# ocamlc -I +compiler-libs -opaque -g -c fl_meta.ml
# File "fl_meta.ml", line 1:
# Error: Invalid import of Fl_metatoken, compiled with -unsafe-string.
#        This compiler has been configured in strict safe-string mode (-force-safe-string)
# make[1]: *** [Makefile:165: fl_meta.cmo] Error 2
# make[1]: Leaving directory '/Users/dan/.opamroot/empty/.opam-switch/build/ocamlfind.1.8.1/src/findlib'
# make: *** [Makefile:13: all] Error 2


#=== ERROR while compiling dune.2.5.0 =========================================#
# context              2.0.6 | macos/x86_64 |  | https://opam.ocaml.org#e2f2c6d2
# path                 ~/.opamroot/empty/.opam-switch/build/dune.2.5.0
# command              ~/.opamroot/opam-init/hooks/sandbox.sh build ocaml bootstrap.ml -j 7
# exit-code            2
# env-file             ~/.opamroot/log/dune-99615-d682c6.env
# output-file          ~/.opamroot/log/dune-99615-d682c6.out
### output ###
# ocamlc -output-complete-exe -w -24 -g -o .duneboot.exe -I boot unix.cma boot/libs.ml boot/duneboot.ml
# File "boot/duneboot.ml", line 1:
# Error: Invalid import of Libs, compiled with -unsafe-string.
#        This compiler has been configured in strict safe-string mode (-force-safe-string)


#=== ERROR while compiling camlp5.7.11 ========================================#
# context              2.0.6 | macos/x86_64 |  | https://opam.ocaml.org#e2f2c6d2
# path                 ~/.opamroot/empty/.opam-switch/build/camlp5.7.11
# command              ~/.opamroot/opam-init/hooks/sandbox.sh build make world.opt
# exit-code            2
# env-file             ~/.opamroot/log/camlp5-99615-e80101.env
# output-file          ~/.opamroot/log/camlp5-99615-e80101.out
### output ###
# [...]
# ocamlc  -c pconfig.mli
# ocamlc  -c pconfig.ml
# File "pconfig.ml", line 1:
# Error: Invalid import of Pconfig, compiled with -unsafe-string.
#        This compiler has been configured in strict safe-string mode (-force-safe-string)
# make[3]: *** [Makefile:22: pconfig.cmo] Error 2
# make[3]: Leaving directory '/Users/dan/.opamroot/empty/.opam-switch/build/camlp5.7.11/ocaml_stuff/4.10.0/utils'
# make[2]: *** [Makefile:9: all] Error 2
# make[2]: Leaving directory '/Users/dan/.opamroot/empty/.opam-switch/build/camlp5.7.11/ocaml_stuff'
# make[1]: *** [Makefile:33: boot/camlp5] Error 2
# make[1]: Leaving directory '/Users/dan/.opamroot/empty/.opam-switch/build/camlp5.7.11'
# make: *** [Makefile:150: world.opt] Error 2



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫 
┌─ The following actions failed
│ λ build camlp5    7.11
│ λ build dune      2.5.0
│ λ build ocamlfind 1.8.1
└─ 
╶─ No changes have been performed
einsteinium$ ocamlc --version
4.10.0
```

--- a/INSTALL_OPAM.md
+++ b/INSTALL_OPAM.md
@@ -13,11 +13,21 @@ First install opam and needed prerequisites:
 
   Then set up the opam system as follows.
 
-  ```bash
-  $ opam init --bare
-  $ opam switch create --empty empty
-  $ opam install -y num lablgtk conf-gtksourceview lablgtk3-sourceview3 camlp5
-  ```
+  - with latest OCaml (not guaranteed to work well with Coq):
+
+    ```bash
+    $ opam init --bare
+    $ opam switch create --empty empty
+    $ opam install -y num lablgtk conf-gtksourceview lablgtk3-sourceview3 camlp5
+    ```
+
+  - with OCaml 4.07.1+flambda (should work on most systems, refer to [#1315](https://github.com/UniMath/UniMath/issues/1315) for details)
+
+    ```bash
+    $ opam init --bare
+    $ opam switch create with-coq 4.07.1+flambda
+    $ opam install -y num lablgtk conf-gtksourceview lablgtk3-sourceview3 camlp5
+    ```
 
 - Under Ubuntu:
 

